### PR TITLE
Feature signup login integration #14

### PR DESCRIPTION
**Pull request by tmclean15**
Access tokens are saved as HttpOnly cookies. Remember me functionality saves email and password in localStorage. Front-end validation is fairly thorough, but edge-cases should be tested in future unit tests.

### CODE REVIEW

File: server/migrations/versions/bff3ff3c07db_.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""empty message
```
Comment (MalcolmAG) [inline server/migrations/versions/bff3ff3c07db_.py]: You should remove the two migrations, as these break "flask db downgrade"

File: client/src/pages/Login.js [hunk 2]
```diff
@@ -47,13 +50,20 @@ const useStyles = makeStyles((theme) => ({
...
   // State variables
   const [invalidEmail, setInvalidEmail] = useState(false);
-  const [invalidPassword, setInvalidPassword] = useState(false);
-  const [rememberMe, setRememberMe] = useState(false);
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
+  const [invalidLogin, setInvalidLogin] = useState(false);
+  const [rememberMe, setRememberMe] = useState(
+    localStorage.getItem("rememberMe") ? true : false
+  );
```
Comment (MalcolmAG) [inline client/src/pages/Login.js:60]: What's the purpose of localStorage? Doesn't the state value do the same thing?

**Review by MalcolmAG — COMMENTED** [review]
I think you should delete the migrations, as they break "flask db downgrade"

File: client/src/pages/Login.js (hunk 2 above)
Comment (tmclean15) [reply client/src/pages/Login.js:60]: This sets the original value of the state variable. If the user chooses "remember me" their credentials will be saved in localStorage, so we want the original value to be whatever is saved in localStorage if it exists.

**Review by MalcolmAG — APPROVED** [review]
Looks good
